flow360/component/simulation/outputs/outputs.py

```diff
@@ -152,14 +153,27 @@ class ProbeOutput(Flow360BaseModel):
     """Probe monitor output settings."""
 
     name: str = pd.Field()
-    entities: Optional[EntityList[Point]] = pd.Field(None, alias="probe_points")
+    entities: Optional[EntityList[Point, PointArray]] = pd.Field(None, alias="probe_points")
     output_fields: UniqueItemList[CommonFieldNames] = pd.Field()
     output_type: Literal["ProbeOutput"] = pd.Field("ProbeOutput", frozen=True)
 
     def load_point_location_from_file(self, file_path: str):
         """Load probe point locations from a file."""
         raise NotImplementedError("Not implemented yet.")
 
+    @pd.field_validator("entities", mode="after")
+    @classmethod
+    def check_entities_type(cls, value):
+        """check to ensure every entity has the same type"""
+        if value is None:
+            return value
+        for entity in value.stored_entities:
```

Comment:
Can you encapsulate this into a function and call it in both classes instead?

Reply:
You can put it in Flow360/flow360/component/simulation/validation/validation_output.py

Reply:
Updated.